.github/workflows/build_branch.yaml

```diff
@@ -0,0 +1,64 @@
+# SPDX-FileCopyrightText: Copyright (c) 2024, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+# SPDX-License-Identifier: Apache-2.0
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: Build dev and main branches
+
+on:
+  push:
+    branches:
+      - 'branch-*'
+      - 'main'
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.event.pull_request.head.label || github.head_ref || github.ref }}'
```

Comment:
I don't know that `github.event.pull_request.head.label` will ever be defined here since this workflow runs on `push` events.

You can view more information about the `github` context here: https://docs.github.com/en/actions/learn-github-actions/contexts#github-context

Reply:
@ajschmidt8 good catch, confirmed neither `github.event.pull_request.head.label` or `github.head_ref` were defined. Only `github.ref` was defined.